src/watcher.js

```diff
@@ -0,0 +1,109 @@
+require('dotenv').config()
+const path = require('path')
+const Web3 = require('web3')
+const { connectWatcherToQueue } = require('./services/amqpClient')
+const { getBlockNumber } = require('./tx/web3')
+const processDeposits = require('./events/processDeposits')
+const processCollectedSignatures = require('./events/processCollectedSignatures')
+const processWithdraw = require('./events/processWithdraw')
+const { redis } = require('./services/redisClient')
+const { getRequiredBlockConfirmations } = require('./tx/web3')
+
+if (process.argv.length < 3) {
+  console.error('Please check the number of arguments, config file was not provided')
+  process.exit(1)
+}
+
+const config = require(path.join('../config/', process.argv[2]))
+
+const provider = new Web3.providers.HttpProvider(config.url)
+const web3Instance = new Web3(provider)
+const bridgeContract = new web3Instance.eth.Contract(config.abi, config.contractAddress)
+const lastBlockRedisKey = `${config.id}:lastProcessedBlock`
+let lastProcessedBlock = 0
+let requiredBlockConfirmations = 1
```

Comment:
@patitonar 
can  we set it from .env file? HOME/FOREIGN_FALLBACK_BLOCK_CONFIRMATION

@akolotov agree?

Reply:
`let requiredBlockConfirmations = 1` works only as initialization, that value will be overwritten by `requiredBlockConfirmations()` call from the contract. Should we add `HOME/FOREIGN_FALLBACK_BLOCK_CONFIRMATION` on .env file that would be used if present instead of the value from the contract?

Reply:
It is assumed that we will always get number of block required for confirmation from the contract so there is no case when `HOME/FOREIGN_FALLBACK_BLOCK_CONFIRMATION` could be used. If we cannot invoke `requiredBlockConfirmations()` it means that something wrong with contracts and we should not continue operating.

In order to get rid of confusion with initialization of the  variable `requiredBlockConfirmations` does it make sense to initialize it as `null` in the line 24 to present that it is a variable declaration rather than a initialization of a value?